nginx/modsecurity/ngx_http_modsecurity.c

```diff
@@ -858,11 +911,20 @@ ngx_http_modsecurity_handler(ngx_http_request_t *r)
     ngx_http_modsecurity_loc_conf_t *cf;
     ngx_http_modsecurity_ctx_t      *ctx;
     ngx_int_t                        rc;
+    ngx_time_t                      *start_time;
+    static ngx_int_t                ret, modsec_latency_ret;
+
+    ngx_time_update();
+    start_time = ngx_timeofday();
 
     cf = ngx_http_get_module_loc_conf(r, ngx_http_modsecurity);
 
     /* Process only main request */
     if (r != r->main || !cf->enable) {
+        modsec_latency_ret = ngx_http_calculate_modsec_latency(r, start_time);
```

Comment:
A lot of this verbosity can be avoided if the handler being set in line 640 can be changed to something like ngx_http_modsecurity_handler_with_timer and define it to set the timing variable and return the output of the current handler as it is.

Reply:
done. Thanks for the tip. It was really useful.